src/jacobians.jl

```diff
@@ -370,8 +370,7 @@ function finite_difference_jacobian!(
                 # Now return x1 back to its original value
                 ArrayInterface.allowed_setindex!(x1, x1_save, color_i)
             else # Perturb along the colorvec vector
-                @. fx1 = x1 * (_color == color_i)
-                tmp = norm(fx1)
+                tmp = norm(x1 .* (_color .== color_i))
```

Comment:
This makes it allocating. Could you create a cache vector that has length `max(fx1, x1)`?

Reply:
That's a good point.  I don't think fx1 matters at all for this particular calculation, it just needs to be length(x1). However,  I'm not sure where you would want to cache it.  Could modify x1 in place, then recopy over the entries in x back into x1.  Or I suppose you could create another vector in JacobianCache.

Reply:
actually couldn't we do something simpler?  Seem like all we need is the one entry in x1 that corresponds to the current color.  Doesn't seem necessary to do the multiplication.  `tmp = abs(x1[_color .== color_i])`  or perhaps I'm missing another use case.

Reply:
scratch that, it still allocates.   The version below doesn't allocate.